lib/core/src/client/preview/start.js

```diff
@@ -62,84 +58,76 @@ function showException(exception) {
   logger.error(exception.stack);
 }
 
-export default function start(render, { decorateStory } = {}) {
-  // check whether we're running on node/browser
-  const isBrowser =
-    navigator &&
-    navigator.userAgent &&
-    navigator.userAgent !== 'storyshots' &&
-    !(navigator.userAgent.indexOf('Node.js') > -1) &&
-    !(navigator.userAgent.indexOf('jsdom') > -1);
-
-  const storyStore = new StoryStore();
-  const reduxStore = createStore(reducer);
-  const context = {
-    storyStore,
-    reduxStore,
-    decorateStory,
-    showMain,
-    showError,
-    showException,
-  };
-
-  const clientApi = new ClientApi(context);
-
-  let channel;
-  if (isBrowser) {
-    // setup preview channel
-    channel = createChannel({ page: 'preview' });
-    channel.on(Events.SET_CURRENT_STORY, data => {
-      reduxStore.dispatch(Actions.selectStory(data.kind, data.story));
-    });
-    addons.setChannel(channel);
-    Object.assign(context, { channel });
+const isBrowser =
```

Comment:
The iframe gets reloaded often when you switched between /settings and /components a lot.

I tried to make this process a bit more optimised.